### ISSUE TITLE
extension for emacs-deferred

### ISSUE THREAD

**Issue by gonewest818** [emacs-elsa/Elsa#170]
Thank you all for your impressive and valuable work here!  

The library [emacs-deferred](https://github.com/kiwanami/emacs-deferred) introduces macros to express deferred chains of asynchronous tasks.  The `deferred:$` macro uses the anaphoric `it`, which it appears requires further explanation for the static analyzer.

Additionally, the author of the library designed it such that, in common usage, people will be writing lambda forms that trigger the notice "You can eta convert the lambda form and use the function directly".  https://github.com/kiwanami/emacs-deferred#basic-usage

**Comment by Fuco1** [comment]
None of the examples should trigger that notice.  It should only be triggered when the function can be reduced:

    (lambda (x) (stringp x)) 

is the same as

    #'stringp

but 

    (lambda (x) (message "foo bar %s" x))

can not be reduced in such a way (x is not a direct argument).

Do you have a repro case?

**Comment by gonewest818** [comment]
Right, sorry, that wasn’t literally the repro case.  See instead https://github.com/gonewest818/geolocation.el/blob/master/geolocation.el <https://github.com/gonewest818/geolocation.el/blob/master/geolocation.el> near line 480 and again at 583.
